Mathlib/Data/List/Indexes.lean

```diff
@@ -168,13 +104,46 @@ theorem map_enumFrom_eq_zipWith : ∀ (l : List α) (n : ℕ) (f : ℕ → α 
         simp only [comp, Nat.add_assoc, Nat.add_comm, Nat.add_succ]
       simp only [length_cons, Nat.succ.injEq] at e; exact e
 
+theorem length_mapIdx_go (f : ℕ → α → β) : ∀ (l : List α) (arr : Array β),
```

Comment:
Flagging for reviewers: these four lemmas are new - the remainder of this PR is
- moving the "old definition" lemmas to a "deprecated" section (and deprecating them)
- the lemma `getElem?_eq`
- minor golfs of proofs (such as `mapIdx_eq_enum_map`); imho for the better

Reply:
Thanks for writing the summary for me!